solutions/USACO jan16-fortmoo.mdx

```diff
@@ -0,0 +1,142 @@
+---
+id: usaco-600
+source: USACO Platinum 2016 January
+title: Fort Moo
+author: Kevin Sheng
+---
+
+[Official Editorial](http://www.usaco.org/current/data/sol_fortmoo_platinum_jan16.html)
+
+The approach here is a bit different from the official solution, using an adaption of 2D Kadane's algorithm instead.
+
+Iterate through all possible pairs of columns. For each pair, we'll solve the problem with the restriction that the fort must be aligned along these two columns. For each row between these two columns, we will use prefix sums to check if the area between the two columns and in the row is clear.
+
+We keep a running variable `r_start` which keeps track of the earliest row we can start the fort at and iterate through all possible row endings (`r_end` in the code). Note that if a swampy area resides in one of the columns themselves, no fort can go through that row, so we must check for that and update `r_start` accordingly.
```

Comment:
maybe use \texttt and latex instead of code for the variables

Reply:
I don't really see any reason to?

Reply:
i think it's convention to; what do others think?

Reply:
yes

Reply:
What's the syntax for Latex again? ```
\texttt{random latex stuff here}
```
Right?

Reply:
yeah, but enclosed in dollar signs, like `$\texttt{r\_start}$` (underscores need to be escaped)